Sage/carbon
# feat(pill): add text wrapping and max width support to component

fix #5192

### Proposed behaviour


**Pill**:
Adds `wrapText` and `maxWidth` props to `Pill` component.
Refactors Pill to be TypeScript

**MultiSelect**:
Sets the `max-width` of the `StyledSelectPillContainer` to 100%, adds  `wrapPillText` prop to allow the text to be wrapped when it would overflow the input

![image](https://user-images.githubusercontent.com/44157880/176705491-c74f105c-523b-465c-b492-92f29983dc66.png)


### Current behaviour


No support for truncation or wrapping in `Pill` and long text strings will cause Pills to overflow input of `MultiSelect`

### Checklist




#### QA


### Additional context



### Testing instructions


Sandbox from linked issue

`select-multiselect--pills-with-long-text` and `pill--wrapped` stories added

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
